broker-core/src/main/java/io/zeebe/broker/exporter/stream/ExporterStreamProcessor.java

```diff
@@ -131,8 +138,6 @@ public void onClose() {
         container.context.getLogger().error("Error on close", e);
       }
     }
-
-    state.close();
```

Comment:
That should work if there's a blocking equivalent (i.e. we don't want to leave the closing phase until the state is closed). I think closing the state should probably be refactored to the stream processor controller anyway since all processors use one and need to do this.

Reply:
> but if an exception would be thrown in state.close(), the actor could still close but the state wouldn't be right? Then runUntilDone() might be better? Though we don't handle errors at all right now, and retrying might not be particularly useful anyway...

Yes could be true. 

> Though we don't handle errors at all right now, and retrying might not be particularly useful anyway... man_shrugging

I totally agree :D How to fix this then :sweat_smile: 

> I anyway think closing the state should probably be refactored to the stream processor controller anyway since all processors use one and need to do this, though I guess it doesn't have to be in this issue in particular.

Yes for sure. I think is ok that we have this in this issue is not a big change :)